### DISCUSSION TITLE
Conebeam projection using torch.tensor

### DISCUSSION THREAD

**Discussion by yongjinColinChoi** [astra-toolbox/astra-toolbox#388] [Q&A] [answered]
I have a question about Conebeam projection using astra-library.

I have a CT volume, and I want to Project it using conebeam geometry like id, projdata = astra.create_sino3d_gpu(cube_id, proj_geom, vol_geom).

here is my question. I want Input CT volume's data type to be torch.tensor and requires_grad = True, so that after conebeam projection I can use torch.autograd. 

Is there any method to project 3D conebeam geometry using input tensors and can track gradient during projection?

**Comment by askorikov** [comment, marked as answer]
You can look into Tomosipo library, which provides integration of Astra Toolbox with PyTorch, specifically: https://aahendriksen.gitlab.io/tomosipo/ref/tomosipo.torch_support.html#tomosipo.torch_support.to_autograd
